### PR TITLE
APIS-7019 Use commands for block and unblock application

### PR DESCRIPTION
**Pull request by petekirby-ee**

### CODE REVIEW

**Review by Tobias1087 — APPROVED** [review]

File: acceptance/test/uk/gov/hmrc/gatekeeper/stubs/ThirdPartyApplicationStub.scala [hunk 1]
```diff
@@ -161,8 +164,9 @@ trait ThirdPartyApplicationStub extends WireMockExtensions with ApplicationWithS
     stubFor(get(urlEqualTo(s"/gatekeeper/application/${applicationId.value.toString()}")).willReturn(aResponse().withBody(defaultApplicationWithHistory.toJsonString).withStatus(OK)))
   }
 
-  def stubApplicationForUnblockSuccess(applicationId: ApplicationId) = {
-    stubFor(post(urlEqualTo(s"/application/${applicationId.value.toString()}/unblock")).willReturn(aResponse().withStatus(OK)))
+  def stubApplicationForUnblockSuccess(applicationId: ApplicationId, gkAppResponse: GKApplicationResponse) = {
+    val response = DispatchSuccessResult(gkAppResponse)
+    stubFor(patch(urlEqualTo(s"/applications/${applicationId.toString()}/dispatch")).willReturn(aResponse().withStatus(OK).withBody(Json.toJson(response).toString())))
```
Comment (johnsgp) [inline acceptance/test/uk/gov/hmrc/gatekeeper/stubs/ThirdPartyApplicationStub.scala:169]: Should this stub expect a body?

File: acceptance/test/uk/gov/hmrc/gatekeeper/stubs/ThirdPartyApplicationStub.scala [hunk 2]
```diff
@@ -187,8 +191,9 @@ trait ThirdPartyApplicationStub extends WireMockExtensions with ApplicationWithS
     stubFor(get(urlPathMatching(s"/submissions/latestiscompleted/.*")).willReturn(aResponse().withStatus(NOT_FOUND)))
   }
 
-  def stubApplicationForBlockSuccess() = {
-    stubFor(post(urlEqualTo(s"/application/${applicationId.value.toString()}/block")).willReturn(aResponse().withStatus(OK)))
+  def stubApplicationForBlockSuccess(applicationId: ApplicationId, gkAppResponse: GKApplicationResponse) = {
+    val response = DispatchSuccessResult(gkAppResponse)
+    stubFor(patch(urlEqualTo(s"/applications/${applicationId.toString()}/dispatch")).willReturn(aResponse().withStatus(OK).withBody(Json.toJson(response).toString())))
```
Comment (johnsgp) [inline acceptance/test/uk/gov/hmrc/gatekeeper/stubs/ThirdPartyApplicationStub.scala:196]: Should this stub expect a body?

File: acceptance/test/uk/gov/hmrc/gatekeeper/stubs/ThirdPartyApplicationStub.scala (hunk 1 above)
Comment (petekirby-ee) [reply acceptance/test/uk/gov/hmrc/gatekeeper/stubs/ThirdPartyApplicationStub.scala]: Done

File: acceptance/test/uk/gov/hmrc/gatekeeper/stubs/ThirdPartyApplicationStub.scala (hunk 2 above)
Comment (petekirby-ee) [reply acceptance/test/uk/gov/hmrc/gatekeeper/stubs/ThirdPartyApplicationStub.scala]: Done

**Review by johnsgp — APPROVED** [review]

**Pull request merged**